### ISSUE TITLE
wrong unreachable, it is reachable

### ISSUE THREAD

**Issue by joshinils** [python/mypy#12161]
**Bug Report**
The following example-bit of code produces an error which does not exist at runtime
```python
a = None
for i in range(2):
    print(i)

    if a is None:
        a = 7
        continue

    print("a =", a)  # this is line 9
    b: int = "b"

```

Error:
```
file.py:9:5: error: Statement is unreachable  [unreachable]
        print("a =", a)
        ^
Found 1 error in 1 file (checked 1 source file)
```

Output:
```
0
1
a = 7
```
**Expected Behavior**
No error reporting.

**Your Environment**

 
- Mypy version used: mypy 0.931
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files):
`.mypy.ini`:
```
# type: ignore

[mypy-requests.*]
ignore_missing_imports = True

[mypy-*/migrations/*]
# Django migrations should not produce any errors:
ignore_errors = True


[mypy]
ignore_missing_imports = False
; disallow_any_expr = True
disallow_any_explicit = True
disallow_untyped_calls = True
disallow_untyped_defs = True
disallow_incomplete_defs = True
check_untyped_defs = True
disallow_untyped_decorators = True
no_implicit_optional = True
warn_redundant_casts = True
warn_unused_ignores = True
warn_no_return = True
warn_return_any = True
warn_unreachable = True
strict_equality = True
show_error_context = True
show_column_numbers = True
show_error_codes = True
pretty = True
show_absolute_path = True 
```
- Python version used: Python 3.8.10
- Operating system and version: Ubuntu 20.04.3 LTS

**Comment by KotlinIsland** [comment]
duplicate of #11612

**Comment by joshinils** [comment]
so it is

> duplicate of #11612
